packages/comparator/src/testcase.ts

```diff
@@ -134,7 +137,7 @@ function createTestcaseComparisonOverview(
       ? (assertions.commonKeys.reduce((acc, v) => acc + v.score!, 0) +
           results.commonKeys.reduce((acc, v) => acc + v.score!, 0)) /
         keysCountCommon
-      : 1,
+      : 0,
```

Comment:
Unrelated fix: If a testcase has no common key but it has new or missing keys, we should not report it as perfect.